backend/src/main/java/be/cytomine/appstore/services/TaskService.java

```diff
@@ -0,0 +1,490 @@
+                throw new TaskServiceException(error);
+            }
+            return Optional.empty();
+        }
+
+        log.info("UploadTask: pushing task image...");
+        DockerImage image =
+            new DockerImage(
+            uploadTaskArchive.getDockerImage(),
+            taskIdentifiers.getImageRegistryCompliantName());
+        try {
+            registryHandler.pushImage(image);
+        } catch (RegistryException e) {
+            try {
+                log.debug("UploadTask: failed to push image to registry");
+                log.debug("UploadTask: attempting to delete storage...");
+                fileStorageHandler.deleteStorage(storage);
+                log.info("UploadTask: storage deleted");
+            } catch (FileStorageException ex) {
+                log.error("UploadTask: file storage service is failing [{}]", ex.getMessage());
+                AppStoreError error = ErrorBuilder
+                    .build(ErrorCode.REGISTRY_PUSHING_TASK_IMAGE_FAILED);
+                throw new TaskServiceException(error);
+            }
+        } finally {
+            uploadTaskArchive.getDockerImage().delete();
+        }
+        log.info("UploadTask: image pushed to registry");
+
+        // save task info
+        Task task = new Task();
+        task.setIdentifier(taskIdentifiers.getLocalTaskIdentifier());
+        task.setStorageReference(taskIdentifiers.getStorageIdentifier());
+        task.setImageName(taskIdentifiers.getImageRegistryCompliantName());
+        task.setName(uploadTaskArchive.getDescriptorFileAsJson().get("name").textValue());
+        task.setNameShort(uploadTaskArchive
+            .getDescriptorFileAsJson()
+            .get("name_short")
+            .textValue());
+        task.setDescriptorFile(
+            uploadTaskArchive.getDescriptorFileAsJson().get("namespace").textValue());
+        task.setNamespace(uploadTaskArchive.getDescriptorFileAsJson().get("namespace").textValue());
+        task.setVersion(uploadTaskArchive.getDescriptorFileAsJson().get("version").textValue());
+        task.setInputFolder(
+            uploadTaskArchive
+            .getDescriptorFileAsJson()
+            .get("configuration")
+            .get("input_folder")
+            .textValue());
+        task.setOutputFolder(
+            uploadTaskArchive
+            .getDescriptorFileAsJson()
+            .get("configuration")
+            .get("output_folder")
+            .textValue());
+
+        // resources
+        JsonNode resources =
+            uploadTaskArchive.getDescriptorFileAsJson().get("configuration").get("resources");
+
+        if (!Objects.nonNull(resources)) {
+            task.setRam(defaultRam);
+            task.setCpus(defaultCpus);
+        } else {
+            task.setRam(resources.path("ram").asText(defaultRam));
+            task.setCpus(resources.path("cpus").asInt(defaultCpus));
+            task.setGpus(resources.path("gpus").asInt(0));
+        }
```

Comment:
remove the `if` and keep the `else` but remove the `defaultRam`, `defaultCpus` in the `else` clause